### PR TITLE
Add limitation in the array of events for the publish endpoint genera…

### PR DESCRIPTION
**Pull request by shudhansu-shekhar**
…teAndPublish

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: <shudhansu-shekhar> <shudhansu.shekhar.ext@ericsson.com>

### CODE REVIEW

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java [hunk 1]
```diff
@@ -78,6 +78,13 @@ public class ProducerController {
     @Value("${activedirectory.publish.enabled}")
     private boolean isAuthenticationEnabled;
 
+    @Value("${maxSizeOfInputArray:250}")
+    private int maxSizeOfInputArray;
+
+    public void setMaxSizeOfInputArray(int maxSizeOfInputArray) {
```
Comment (z-sztrom) [inline publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java:84]: Where the method is used?

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java [hunk 2]
```diff
@@ -303,6 +310,14 @@ public ResponseEntity generateAndPublish(final String msgProtocol, final String
         if (bodyJson.isJsonObject()) {
             events.add(getAsJsonObject(bodyJson));
         } else if (bodyJson.isJsonArray()) {
+            //here add check for limitation for events in array is fetched from REMReM property and checked during publishing.
+            if (bodyJson.getAsJsonArray().size() > maxSizeOfInputArray) {
+                return createResponseEntity(HttpStatus.BAD_REQUEST, JSON_ERROR_STATUS,
+                        "The number of events in the input array is exceeded the allowed limit of 250 events. " +
```
Comment (z-sztrom) [inline publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java:316]: As for generate, i.e. hard-coded value.

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java [hunk 3]
```diff
@@ -303,6 +310,14 @@ public ResponseEntity generateAndPublish(final String msgProtocol, final String
...
+                                "This issue occurred because the input array contains more than 250 events, which is not supported by the system. "
+                                + "To resolve this, please divide the events in to smaller arrays, ensuring each array contains no more than 250 events,"
+                                + "and try to publish them again. This limitation helps to maintain system performance and stability.");
+            }
             for (JsonElement element : bodyJson.getAsJsonArray()) {
```
Comment (z-sztrom) [inline publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java:321]: Why to call getAsJsonArray() multiple time?

File: publish-service/src/main/resources/application.properties [hunk 4]
```diff
@@ -44,5 +44,7 @@ activedirectory.connectionTimeOut:
 
 spring.mvc.pathmatch.matching-strategy: ANT_PATH_MATCHER
 
+#Maximum size of events in array is fetched from REMReM property and checked during publishing.
```
Comment (z-sztrom) [inline publish-service/src/main/resources/application.properties:47]: As for generate.

**Review by z-sztrom — CHANGES_REQUESTED** [review]

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java (hunk 1 above)
Comment (shudhansu-shekhar) [reply publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java:84]: Remove it

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java (hunk 2 above)
Comment (shudhansu-shekhar) [reply publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java:316]: Done

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java (hunk 3 above)
Comment (shudhansu-shekhar) [reply publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java:321]: fix it

**Review by shudhansu-shekhar — COMMENTED** [review]
Done, update the code as suggest

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java [hunk 5]
```diff
@@ -303,7 +306,14 @@ public ResponseEntity generateAndPublish(final String msgProtocol, final String
         if (bodyJson.isJsonObject()) {
             events.add(getAsJsonObject(bodyJson));
         } else if (bodyJson.isJsonArray()) {
-            for (JsonElement element : bodyJson.getAsJsonArray()) {
+            JsonArray bodyJsonArray = bodyJson.getAsJsonArray();
+            //here add check for limitation for events in array is fetched from REMReM property and checked during publishing.
+            if (bodyJsonArray.size() > maxSizeOfInputArray) {
+                return createResponseEntity(HttpStatus.BAD_REQUEST, JSON_ERROR_STATUS,
+                        "The number of events in the input array is too high: " + bodyJsonArray.size() + " > " + maxSizeOfInputArray + ", " +
```
Comment (z-sztrom) [inline publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java:313]: -> maxSizeOfInputArray + "; " + "you ... to increase it."

**Review by z-sztrom — CHANGES_REQUESTED** [review]

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java (hunk 5 above)
Comment (shudhansu-shekhar) [reply publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java:313]: Done

**Review by shudhansu-shekhar — COMMENTED** [review]
Done, please check latest code

**Review by z-sztrom — APPROVED** [review]

**Pull request merged**